kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExport.kt

```diff
@@ -52,6 +52,7 @@ internal fun produceObjCExportInterface(
     val disableSwiftMemberNameMangling = config.configuration.getBoolean(BinaryOptions.objcExportDisableSwiftMemberNameMangling)
     val ignoreInterfaceMethodCollisions = config.configuration.getBoolean(BinaryOptions.objcExportIgnoreInterfaceMethodCollisions)
     val reportNameCollisions = config.configuration.getBoolean(BinaryOptions.objcExportReportNameCollisions)
+    val errorOnNameCollisions = config.configuration.getBoolean(BinaryOptions.objcExportErrorOnNameCollisions)
```

Comment:
Here and in other places where these two parameters are handled, it makes sense to replace them with a single enum-typed parameter.

Reply:
Thanks!

> but left them as boolean cli flags for backwards compatibility.

Considering that the previous flag was introduced in [Kotlin 2.0.0-Beta2](https://youtrack.jetbrains.com/issue/KT-63153/Native-implement-a-flag-to-emit-compiler-warnings-on-ObjCExport-name-collisions), and we are going to merge this PR to Kotlin 2.0.0-Beta5, there is no need for keeping backward compatibility :)